app/src/main/java/com/example/androidtrainingproject/ui/profile/UserProfile.kt

```diff
@@ -0,0 +1,98 @@
+import com.example.androidtrainingproject.ui.destinations.OrderHistoryDestination
+import com.example.androidtrainingproject.ui.shared.AppTopBar
+import com.example.androidtrainingproject.ui.shared.WideButton
+import com.example.androidtrainingproject.ui.theme.Gray
+import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+
+@Destination
+@Composable
+fun UserProfile(destinationsNavigator: DestinationsNavigator) {
+    Scaffold(
+        topBar = {
+            AppTopBar(
+                middleText =  stringResource(id = R.string.user_profile),
+                destinationsNavigator = destinationsNavigator,
+                backIcon = true
+            )
+        }
+    ) { paddingValues ->
+        Column(
+            Modifier
+                .fillMaxSize()
+                .paint(
+                    painter = painterResource(id = R.drawable.app_background),
+                    contentScale = ContentScale.FillBounds
+                )
+                .padding(top = paddingValues.calculateTopPadding())
+                .padding(dimensionResource(id = R.dimen.padding_m))
+        ) {
+            Image(
+                painter = painterResource(id = R.drawable.profile_picture),
+                contentDescription = null,
+                modifier = Modifier
+                    .align(Alignment.CenterHorizontally)
+                    .padding(
+                        top = dimensionResource(id = R.dimen.padding_xxl),
+                        bottom = dimensionResource(id = R.dimen.padding_m_plus)
+                    )
+            )
+            Text(
+                modifier = Modifier.align(Alignment.CenterHorizontally),
+                text = GlobalStorage.userEmail.value,
+                fontWeight = FontWeight.Bold
+            )
+            Divider(
+                modifier = Modifier.padding(vertical = dimensionResource(id = R.dimen.padding_m)),
+                color = Gray,
+                thickness = dimensionResource(id = R.dimen.divider_size)
+            )
+            Row(
+                modifier = Modifier.clickable { destinationsNavigator.navigate(OrderHistoryDestination) }
+            ) {
+                Text(text = stringResource(id = R.string.order_history))
+                Spacer(modifier = Modifier.weight(0.5f))
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_chevron_right),
+                    contentDescription = null
+                )
+            }
+            WideButton(
+                modifier = Modifier.padding(top = dimensionResource(id = R.dimen.padding_xxl)),
+                text = stringResource(id = R.string.logout),
+                enabled = true,
+                onClick = {
+                    APIClient.clearJwt()
+                    destinationsNavigator.navigate(LoginScreenDestination)
```

Comment:
Unfortunately, this won't be enough. All the previous screens will still be in your backstack. It needs to be cleared. Check what functions does `destinationsNavigator` offer for this and experiment as, unfortunately, it isn't documented anywhere. Ping us if you need assistance!

Tip: In order to test if you've cleared the backstack you can add a temporary test TopBar to the Login screen with a back button calling `navigateUp`. If it backs to a previous screen successfully, then you have not succeeded in clearing the backstack.